core/qthelper.cpp

```diff
@@ -560,15 +559,29 @@ QLocale getLocale()
 	return loc;
 }
 
-void set_filename(const char *filename, bool force)
+void set_filename(const FileLocation &location, bool force)
 {
-	if (!force && existing_filename)
+	if (!force && !currentFile.isNone())
 		return;
-	free((void *)existing_filename);
-	if (filename)
-		existing_filename = strdup(filename);
-	else
-		existing_filename = NULL;
+	currentFile = location;
+}
+
+void set_current_file_none()
+{
+	currentFile = FileLocation();
+}
+
+char *get_current_file_name()
+{
+	return strdup(qPrintable(currentFile.getName()));
+}
```

Comment:
The mixing of C code and C++ code here is just strange.
Maybe this is the right thing to do, but consider me not convinced

Reply:
This function also had to be introduced temporarily owing to multiple commits and is unused at the end of the series. Thanks for noting!